### PR TITLE
Feature/df 2987 improve logging for jwt saml tokens

### PR DESCRIPTION
**Pull request by birom**

### CODE REVIEW

**Review by palsch — APPROVED** [review]

File: jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/config/WebConfigurer.java [hunk 1]
```diff
@@ -169,6 +171,12 @@ private void initMDCInserting(ServletContext servletContext, EnumSet<DispatcherT
 
         mdcInsertingFilter.addMappingForUrlPatterns(disps, true, "/*");
         mdcInsertingFilter.setAsyncSupported(true);
+
```
Comment (seco-mgabor) [inline jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/config/WebConfigurer.java:174]: What is this filter for? Please explain it in a comment.

File: jobroom-gateway-app/src/main/resources/config/bootstrap.yml [hunk 2]
```diff
@@ -27,3 +27,7 @@ spring:
                 initial-interval: 1000
                 max-interval: 2000
                 max-attempts: 100
+
+logging:
+    pattern:
```
Comment (seco-mgabor) [inline jobroom-gateway-app/src/main/resources/config/bootstrap.yml:32]: I don't understand the next line, sorry.

File: jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/infrastructure/logging/CustomMDCInsertingServletFilter.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+import javax.servlet.http.Cookie;
+import javax.servlet.http.HttpFilter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+
```
Comment (seco-mgabor) [inline jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/infrastructure/logging/CustomMDCInsertingServletFilter.java:17]: JavaDoc is missing. What's the purpouse of this filter?

File: jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/jwt/TokenProvider.java [hunk 4]
```diff
@@ -44,6 +44,8 @@ public TokenProvider(JHipsterProperties jHipsterProperties, ClaimMapper claimMap
     }
 
     public String createToken(Authentication authentication, boolean rememberMe) {
+        LOGGER.debug("Creating Token for authentication: {}", authentication);
```
Comment (seco-mgabor) [inline jobroom-gateway-app/src/main/java/ch/admin/seco/jobroom/security/jwt/TokenProvider.java:47]: Are you sure that ExtId, Email Address and the Session ID are visible in the log afterwards?

**Review by seco-mgabor — CHANGES_REQUESTED** [review]

**Review by seco-mgabor — APPROVED** [review]

**Pull request merged**